components/brave_wallet/browser/eth_data_parser.cc

```diff
@@ -80,66 +38,149 @@ bool GetTransactionInfoFromData(const std::string& data,
+      return false;
+  } else if (selector == "0x095ea7b3") {
+    *tx_type = mojom::TransactionType::ERC20Approve;
+    if (!brave_wallet::ABIDecode({"address", "uint256"}, calldata, tx_params,
+                                 tx_args))
       return false;
-    if (!GetUint256HexFromData(std::string(left_over_data), &value,
-                               &left_over_data))
+  } else if (selector == "0x23b872dd") {
+    *tx_type = mojom::TransactionType::ERC721TransferFrom;
+    if (!brave_wallet::ABIDecode({"address", "address", "uint256"}, calldata,
+                                 tx_params, tx_args))
       return false;
-    // Very strictly must have correct data
-    if (left_over_data.length() > 0)
+  } else if (selector == "0x42842e0e") {
+    *tx_type = mojom::TransactionType::ERC721SafeTransferFrom;
+    if (!brave_wallet::ABIDecode({"address", "address", "uint256"}, calldata,
+                                 tx_params, tx_args))
+      return false;
+  } else if (selector == "0x3598d8ab") {
+    *tx_type = mojom::TransactionType::ETHSwap;
+
+    // Function:
+    // sellEthForTokenToUniswapV3(bytes encodedPath,
+    //                            uint256 minBuyAmount,
+    //                            address recipient)
+    //
+    // Ref:
+    // https://github.com/0xProject/protocol/blob/b46eeadc64485288add5940a210e1a7d0bcb5481/contracts/zero-ex/contracts/src/features/interfaces/IUniswapV3Feature.sol#L29-L41
+    if (!brave_wallet::ABIDecode({"bytes", "uint256", "address"}, calldata,
+                                 tx_params, tx_args))
       return false;
 
-    if (tx_args) {
-      tx_args->push_back(address);
-      tx_args->push_back(value);
+    std::vector<std::string> decoded_path;
+    if (!brave_wallet::UniswapEncodedPathDecode(tx_args->at(0), &decoded_path))
+      return false;
+    std::string fill_path = "0x";
+    for (const auto& path : decoded_path) {
+      fill_path += path.substr(2);
     }
 
-    if (tx_params) {
-      tx_params->push_back("address");
-      tx_params->push_back("uint256");
-    }
-  } else if (*tx_type == mojom::TransactionType::ERC721TransferFrom ||
-             *tx_type == mojom::TransactionType::ERC721SafeTransferFrom) {
-    std::string from, to, token_id;
-    std::string left_over_data = data.substr(10);
-    // Intentional copy of left_over_data
-    if (!GetAddressArgFromData(std::string(left_over_data), &from,
-                               &left_over_data))
-      return false;
-    if (!GetAddressArgFromData(std::string(left_over_data), &to,
-                               &left_over_data))
+    // Populate ETHSwap tx_params and tx_args.
+    *tx_params = {
+        "bytes",    // fill path
+        "uint256",  // maker amount
+        "uint256"   // taker amount
+    };
+    *tx_args = {fill_path,
+                "",  // maker asset is ETH, amount is txn value
+                tx_args->at(1)};
+  } else if (selector == "0x803ba26d" || selector == "0x6af479b2") {
```

Comment:
Can we have such selectors as constants somewhere?

Reply:
What would be a good place to add these constants? `eth_data_parser.cc`? `brave_wallet_constants.h`?

Reply:
Fixed here: d5cf0e6b67bff946c249234866da648c24b9b12a